frontend/src/pages/assignStudents.tsx

```diff
@@ -3,8 +3,32 @@ import { Col, Row } from "react-bootstrap";
 import NavBar from "../components/navBar";
 import styles from "../styles/pageGrids.module.css";
 import { StudentList } from "../components/studentList";
+import ProjectAsignmentList from "../components/project_assignment/projectAssignmentList";
+import { useState } from "react";
+import AssignmentModal from "../components/project_assignment/assignmentModal";
+
+type SkillInfo = { skillName: string; skillColor: string; skillUrl: string };
+export type DropHandler = (
+    studentName: string,
+    studentUrl: string,
+    skillInfo: SkillInfo,
+    projectName: string
+) => void;
 
 const AssignStudentsPage: NextPage = () => {
+    const [showModal, setShowModal] = useState(false);
+    const [modalInfo, setModalInfo] = useState<{
+        studentName: string;
+        studentUrl: string;
+        skillInfo: SkillInfo;
+        projectName: string;
+    }>();
```

Comment:
I think this can also be undefined? Since you don't provide a default value?

```suggestion
    const [modalInfo, setModalInfo] = useState<{
        studentName: string;
        studentUrl: string;
        skillInfo: SkillInfo;
        projectName: string;
    } | undefined>();
```